### PR TITLE
QgsRasterAnalysisUtils: fix range loop constructs warning

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description

Loops create a copy from map. Use reference type to prevent copying.

### CODE REVIEW

File: src/analysis/processing/qgsrasteranalysisutils.cpp [hunk 1]
```diff
@@ -401,7 +401,7 @@ double QgsRasterAnalysisUtils::majorityFromCellValues( std::vector<double> &cell
     int maxCount = 0;
     bool multipleMajorities = false;
     double result = noDataValue;
-    for ( const auto pair : map )
+    for ( const auto &pair : map )
```
Comment (elpaso) [inline src/analysis/processing/qgsrasteranalysisutils.cpp:404]:
```suggestion
    for ( const auto &pair : std::as_const( map ) )
```

File: src/analysis/processing/qgsrasteranalysisutils.cpp [hunk 2]
```diff
@@ -449,7 +449,7 @@ double QgsRasterAnalysisUtils::minorityFromCellValues( std::vector<double> &cell
     int minCount = stackSize;
     bool multipleMinorities = false;
     double result = noDataValue; //result will stay NoData if no minority value exists
-    for ( const auto pair : map )
+    for ( const auto &pair : map )
```
Comment (elpaso) [inline src/analysis/processing/qgsrasteranalysisutils.cpp:452]:
```suggestion
    for ( const auto &pair : std::as_const( map ) )
```

**Review by elpaso — APPROVED** [review]
std::as_const may be a noop but it makes it explicit

**Comment by lbartoletti** [comment]
Right. Done.

**Pull request merged**